src/reportError.ts

```diff
@@ -0,0 +1,71 @@
+import chalk = require('chalk')
+import {Log} from 'pnpm-logger'
+
+export default function reportError (logObj: Log) {
+  if (logObj['err']) {
+    const err = <Error & { code: string }>logObj['err']
+    switch (err.code) {
+      case 'UNEXPECTED_STORE':
+        reportUnexpectedStore(err, logObj['message'])
+        return
+      case 'STORE_BREAKING_CHANGE':
+        reportStoreBreakingChange(err, logObj['message'])
+        return
+      case 'MODULES_BREAKING_CHANGE':
+        reportModulesBreakingChange(err, logObj['message'])
+        return
+      default:
+        printErrorSummary(err.message || logObj['message'])
+        return
+    }
+  }
+  printErrorSummary(logObj['message'])
+}
+
+function reportUnexpectedStore (err: Error, msg: Object) {
+  printErrorSummary(err.message)
+  console.log()
+  console.log(`expected: ${chalk.yellow(msg['expectedStorePath'])}`)
+  console.log(`actual: ${chalk.yellow(msg['actualStorePath'])}`)
+  console.log()
+  console.log(`If you want to use the new store, run the same command with the ${chalk.yellow('--force')} parameter.`)
+}
+
+function reportStoreBreakingChange (err: Error, msg: Object) {
+  printErrorSummary(`The store used for the current node_modules is incomatible with the current version of pnpm`)
+  console.log(`Store path: ${chalk.gray(msg['storePath'])}`)
```

Comment:
You might want to use [outdent](https://www.npmjs.com/package/outdent) package to format multiline strings